src/error/GraphQLError.ts

```diff
@@ -20,6 +20,41 @@ export interface GraphQLErrorExtensions {
   [attributeName: string]: unknown;
 }
 
+interface GraphQLErrorArgs {
+  nodes?: ReadonlyArray<ASTNode> | ASTNode | null;
+  source?: Maybe<Source>;
+  positions?: Maybe<ReadonlyArray<number>>;
+  path?: Maybe<ReadonlyArray<string | number>>;
+  originalError?: Maybe<Error & { readonly extensions?: unknown }>;
+  extensions?: Maybe<GraphQLErrorExtensions>;
+}
+
+type BackwardsCompatibleArgs =
+  | [args?: GraphQLErrorArgs]
+  | [
+      nodes?: GraphQLErrorArgs['nodes'],
+      source?: GraphQLErrorArgs['source'],
+      positions?: GraphQLErrorArgs['positions'],
+      path?: GraphQLErrorArgs['path'],
+      originalError?: GraphQLErrorArgs['originalError'],
+      extensions?: GraphQLErrorArgs['extensions'],
+    ];
+
+function toNormalizedArgs(args: BackwardsCompatibleArgs): GraphQLErrorArgs {
+  const firstArg = args[0];
+  if (firstArg == null || 'kind' in firstArg || 'length' in firstArg) {
```

Comment:
I used `'length' in firstArg` instead of `Array.isArray` here as the latter did not properly discriminate the type (https://stackoverflow.com/a/57343333)

Reply:
Not ideal, but we can leave with it since it temporary code anyway.